pumps/graph_mongo.go

```diff
@@ -0,0 +1,145 @@
+package pumps
+
+import (
+	"context"
+	"strings"
+
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/mitchellh/mapstructure"
+	"github.com/sirupsen/logrus"
+)
+
+const mongoGraphPrefix = "mongo-graph-pump"
+
+type GraphMongoPump struct {
+	CommonPumpConfig
+	MongoPump
+}
+
+func (g *GraphMongoPump) New() Pump {
+	return &GraphMongoPump{}
+}
+
+func (g *GraphMongoPump) GetEnvPrefix() string {
+	return g.dbConf.EnvPrefix
+}
+
+func (g *GraphMongoPump) GetName() string {
+	return "MongoDB Graph Pump"
+}
+
+func (g *GraphMongoPump) Init(config interface{}) error {
+	g.dbConf = &MongoConf{}
+	g.log = log.WithField("prefix", mongoGraphPrefix)
+	g.MongoPump.CommonPumpConfig = g.CommonPumpConfig
+
+	err := mapstructure.Decode(config, &g.dbConf)
+	if err == nil {
+		err = mapstructure.Decode(config, &g.dbConf.BaseMongoConf)
+		g.log.WithFields(logrus.Fields{
+			"url":             g.dbConf.GetBlurredURL(),
+			"collection_name": g.dbConf.CollectionName,
+		}).Info("Init")
+		if err != nil {
+			panic(g.dbConf.BaseMongoConf)
+		}
+	}
+	if err != nil {
+		g.log.Fatal("Failed to decode configuration: ", err)
+	}
+
+	if g.dbConf.MaxInsertBatchSizeBytes == 0 {
+		g.log.Info("-- No max batch size set, defaulting to 10MB")
+		g.dbConf.MaxInsertBatchSizeBytes = 10 * MiB
+	}
+
+	if g.dbConf.MaxDocumentSizeBytes == 0 {
+		g.log.Info("-- No max document size set, defaulting to 10MB")
+		g.dbConf.MaxDocumentSizeBytes = 10 * MiB
+	}
+
+	g.connect()
+
+	g.capCollection()
+
+	indexCreateErr := g.ensureIndexes()
+	if indexCreateErr != nil {
+		g.log.Error(indexCreateErr)
+	}
+
+	g.log.Debug("MongoDB DB CS: ", g.dbConf.GetBlurredURL())
+	g.log.Debug("MongoDB Col: ", g.dbConf.CollectionName)
+
+	g.log.Info(g.GetName() + " Initialized")
+
+	return nil
+}
+
+func (g *GraphMongoPump) WriteData(ctx context.Context, data []interface{}) error {
+	collectionName := g.dbConf.CollectionName
+	if collectionName == "" {
+		g.log.Fatal("No collection name!")
```

Comment:
`WriteData` function has an error type as a return parameter. Does the caller of `WriteData` method log this error? Returning an error if the collection name is empty looks more idiomatic to me.